packages/assets-controllers/src/AssetsContractController.ts

```diff
@@ -93,6 +94,7 @@ export class AssetsContractController extends BaseController<
       chainId: initialChainId,
       onPreferencesStateChange,
       onNetworkStateChange,
+      getNetworkClientById,
```

Comment:
I was thinking about how many controllers will need to do this, maybe we should expose this method over the controller messenger so that the order of creation of the controllers doesn't matter.

Reply:
Did you want to apply this pattern to the other controllers in core? If so, can we address this in a separate PR?

Reply:
will do as part of this PR

Reply:
AssetsContractController is V1 BaseController. So will have to follow this dependency injection pattern for now still